.gitignore

```
@@ -7,6 +7,7 @@
 
 # Packages
 *.egg
+.eggs
```

What makes `.eggs`?

---

It's a funny one - setuptools uses it as part of the PEP 517 procedure, but only when you call `python setup.py ...` directly, and it needs to install a `setup_requires` requirement.  If you use a "regular" PEP517 builder via pip, it'll install the temporary build things in a venv in a temporary directory (which sucks a little because it means Cython invariably detects a change in the numpy version you're building against, and rebuilds everything).

The thing is, that if you run `python setup.py ...` directly, the first lines of the files import all the third-party imports required in `setup_requires`, so it should just report a Python error.  However, if you have a version of numpy, scipy or cython that violates the _version_ requirements, `setup.py` will download suitable versions once it hits `setuptools.setup()`.  The problem is that the damage is already done for us at that point - Cython will have cythonised and compiled against a bad version of Numpy, or using an old version of Cython.

Still, it's possible for the directory to be created, and we will never want it committed to source control.

---

Tx for the explanation -- let's keep it.